src/components/BalancePage/GoalsEditDialog/GoalsEditDialog.tsx

```diff
@@ -0,0 +1,115 @@
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  TextField,
+  DialogActions,
+  Button,
+  MenuItem,
+} from '@mui/material'
+import { SubmitHandler, useForm } from 'react-hook-form'
+import { useDispatch } from 'react-redux'
+import {
+  Goal,
+  createActionDeleteGoal,
+  createActionGoalAdd,
+} from '../../../state/accountConfig'
+
+interface GoalDialogProps {
+  isOpen: boolean
+  update: any
+  goal: Goal
+  updateGoal: any
+}
+
+const GoalEditDialog = (editDialog: GoalDialogProps) => {
+  const dispatch = useDispatch()
+
+  const { register, handleSubmit, reset } = useForm<Goal>({
+    defaultValues: {
+      name: editDialog.goal.name,
+      priority: editDialog.goal.priority,
+      value: editDialog.goal.value,
+      deadline: editDialog.goal.deadline,
+    },
+  })
+
+  const onSubmit: SubmitHandler<any> = async (data) => {
+    dispatch(createActionDeleteGoal(editDialog.goal))
+    dispatch(createActionGoalAdd(data))
```

Comment:
Would be better to have edit action instead of delete & add, but we can leave it like that for now